### PR TITLE
test: improve `*_info` modules tests

### PR DESCRIPTION
**Pull request by jooola**
##### SUMMARY

- Implement the new testing framework (prepare.yml/cleanup.yml #239)
- Fix some uncovered test scenarios (related to #298).
- Structure all the *_info tests using the following structure:
  ```
  gather all
  [gather all with custom options...]
  gather all in check mode
  gather with id
  gather with name
  gather with labels
  [gather with custom option...]
  ```

### CODE REVIEW

File: tests/integration/targets/hcloud_certificate_info/defaults/main/main.yml
Comment (apricote) [inline tests/integration/targets/hcloud_certificate_info/defaults/main/main.yml:4]: Can we delete the certificate after merging or are there any other tests using it?

File: tests/integration/targets/hcloud_floating_ip_info/tasks/test.yml
Comment (apricote) [inline tests/integration/targets/hcloud_floating_ip_info/tasks/test.yml:56]: Why are these commented out?

File: tests/integration/targets/hcloud_image_info/tasks/test.yml
Comment (apricote) [inline tests/integration/targets/hcloud_image_info/tasks/test.yml:104]: This test might be flaky, `key!=value` applies to nearly all snapshots. Either because another test is currently creating a snapshot or when running this locally against a non-clean project.

We could check that our snapshot is not in the list of results, instead of checking `count == 0`

Might also be something for a follow up PR, as the test was included like this in the code before and multiple other tests have the same pattern.

File: tests/integration/targets/hcloud_server_info/tasks/test.yml
Comment (apricote) [inline tests/integration/targets/hcloud_server_info/tasks/test.yml:10]: Could be
```suggestion
      - result.hcloud_server_info | list | count >= 2
```

as we know that we have created 2 servers in `prepare.yml`, but really does not matter much.

File: tests/integration/targets/hcloud_server_info/tasks/test.yml
Comment (apricote) [inline tests/integration/targets/hcloud_server_info/tasks/test.yml:89]: What exactly is "Gather hcloud_server_info (without ip)" this testing?

File: tests/integration/targets/hcloud_certificate_info/defaults/main/main.yml
Comment (jooola) [reply tests/integration/targets/hcloud_certificate_info/defaults/main/main.yml:4]: The target is the only place where this cert is used, in addition it is created and destroyed on each run. This might have been a source of race between parallel tests, as they all used the same cert name.

File: tests/integration/targets/hcloud_floating_ip_info/tasks/test.yml
Comment (jooola) [reply tests/integration/targets/hcloud_floating_ip_info/tasks/test.yml:56]: Because we don't implement it yet https://github.com/ansible-collections/hetzner.hcloud/issues/302

I have a branch with the implementation ready, just waiting on this PR to be merged to open a new PR.

File: tests/integration/targets/hcloud_image_info/tasks/test.yml
Comment (jooola) [reply tests/integration/targets/hcloud_image_info/tasks/test.yml:104]: Yes, I'd rather postpone this change to another PR. We can then also make batch changes to all the `_info` targets.

File: tests/integration/targets/hcloud_server_info/tasks/test.yml
Comment (jooola) [reply tests/integration/targets/hcloud_server_info/tasks/test.yml:89]: The test is part of a bug fix: https://github.com/ansible-collections/hetzner.hcloud/commit/6a7088cf42dcbf01d3c65617db93bff932752cb3

**Review by apricote — APPROVED** [review]

**Pull request merged**
